hpat/datatypes/hpat_pandas_stringmethods_functions.py

```diff
@@ -0,0 +1,238 @@
+# OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+# EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *****************************************************************************
+
+"""
+
+| :class:`pandas.core.strings.StringMethods` functions and operators implementations in HPAT
+
+    .. only:: developer
+
+    This is autogenerated sources for all Unicode string functions supported by Numba.
+    Currently tested 45 functions only. List of functions obtained automatically from
+    `numba.types.misc.UnicodeType` class
+
+    Example of the generated method (for method upper()):
+    `hpat_pandas_stringmethods_upper_parallel_impl` is paralell version
+    (required additional import mentioned in the body)
+
+    @overload_method(StringMethodsType, 'upper')
+    def hpat_pandas_stringmethods_upper(self):
+        _func_name = 'Method stringmethods.upper().'
+
+        if not isinstance(self, StringMethodsType):
+            raise TypingError('{} The object must be a pandas.core.strings. Given: {}'.format(_func_name, self))
+
+        def hpat_pandas_stringmethods_upper_parallel_impl(self):
+            from numba.parfor import (init_prange, min_checker, internal_prange)
+
+            init_prange()
+            result = []
+            item_count = len(self._data)
+            min_checker(item_count)
+            for i in internal_prange(item_count):
+                item = self._data[i]
+                item_method = item.upper()
+                result.append(item_method)
+
+            return pandas.Series(result)
+
+        return hpat_pandas_stringmethods_upper_parallel_impl
+
+        def hpat_pandas_stringmethods_upper_impl(self):
+            result = []
+            item_count = len(self._data)
+            for i in range(item_count):
+                item = self._data[i]
+                item_method = item.upper()
+                result.append(item_method)
+
+            return pandas.Series(result)
+
+        return hpat_pandas_stringmethods_upper_impl
+
+    Test: python -m hpat.runtests hpat.tests.test_hiframes.TestHiFrames.test_str_split_filter
+
+"""
+
+
+import numpy
+import pandas
+
+import numba
+from numba import types
+from numba.extending import overload_method
+from numba.errors import TypingError
+
+from hpat.datatypes.hpat_pandas_stringmethods_types import StringMethodsType
+from hpat.str_arr_ext import to_string_list
+
+
+_hpat_pandas_stringmethods_autogen_global_dict = {'pandas': pandas,
+                                                  'numpy' : numpy,
+                                                  'numba' : numba,
+                                                  'StringMethodsType': StringMethodsType}
+
+_hpat_pandas_stringmethods_functions_params = {
+    'cat': ', others=None, sep=None, na_rep=None, join="left"',
```

Comment:
Why do you need comma at the start? It's easier to split the parameters without the comma in the beginning.